tools/retar.py

```diff
@@ -0,0 +1,54 @@
+"""
+Un-targz and retargz a targz file to ensure reproducible build. 
```

Comment:
Wait, we have to do this ourselves? I thought when you opened this PR that setuptools had gained the ability to make properly reproducible builds itself. I like the idea of reproducible builds, but if it's something we need custom tooling for, I wouldn't bother. I want our release process to be less special, not more special.

IPython doesn't have a particular need for reproducible builds beyond what packages in general have. In fact, it's arguably less important for pure Python packages, because you can unpack them and diff the source tree against that from the git tag.

Reply:
I though too as well; but it's actually just for Sdist. 
We can wait; but that will require changes in core CPython distutils, gzip and tarfile module.
Basically tar does not pass kwargs as options to gzip

So if we want to have that only using core python then we need to wait for 3.7; maybe 3.8

I'm happy to not get that in; but at least this give us a minimal idea of what is missing in core Python; I can try to start a thread on distutils-sig if you like; but with the usual bikesheding...

Reply:
I had a look at this for flit at some point - I think it's doable without changes to the stdlib, if  the packaging tool (flit/setuptools) duplicates a few lines of code from `tarfile` so that it creates the gzip object itself.